server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java

```diff
@@ -107,16 +138,36 @@ public void delete(List<String> names) {
         }
     }
 
+    public void deleteGenerations(Set<Long> generations) {
+        for (Long generation : generations) {
+            generationalFilesTransferTracker.remove(generation);
+        }
+    }
+
     public boolean uploaded(String file) {
         return fileTransferTracker.get(file) == TransferState.SUCCESS;
     }
 
-    public Set<TransferFileSnapshot> exclusionFilter(Set<TransferFileSnapshot> original) {
+    public boolean translogGenerationUploaded(Long generation) {
```

Comment:
does this method need to be public?

Reply:
revised Access modifiers for functions in this class.